byteparse/shared/src/main/scala/byteparse/classparse/ClassParser.scala

```diff
@@ -0,0 +1,456 @@
+        )
+
+        InvokeDynamic(name, descriptor, info.bootstrapMethodAttrIndex)
+      }
+    }
+
+
+    private def getFlag(f: Byte, mask: Int) = (f & mask) != 0
+
+    case class InnerClassFlags(accPublic: Boolean,   accPrivate: Boolean,   accProtected: Boolean,
+                               accStatic: Boolean,   accFinal: Boolean,     accInterface: Boolean,
+                               accAbstract: Boolean, accSynthetic: Boolean, accAnnotation: Boolean,
+                               accEnum: Boolean){
+
+      def this(fs: Array[Byte]) = this(getFlag(fs(1), 0x01), getFlag(fs(1), 0x02), getFlag(fs(1), 0x04),
+                                       getFlag(fs(1), 0x08), getFlag(fs(1), 0x10), getFlag(fs(0), 0x02),
+                                       getFlag(fs(0), 0x04), getFlag(fs(0), 0x10), getFlag(fs(0), 0x20),
+                                       getFlag(fs(0), 0x40))
+    }
+
+
+    object InnerClassFlags {
+      def apply(fs: Array[Byte]) = new InnerClassFlags(fs)
+    }
+
+    case class FieldFlags(accPublic: Boolean,    accPrivate: Boolean,   accProtected: Boolean,
+                          accStatic: Boolean,    accFinal: Boolean,     accVolatile: Boolean,
+                          accTransient: Boolean, accSynthetic: Boolean, accEnum: Boolean) {
+
+      def this(fs: Array[Byte]) = this(getFlag(fs(1), 0x01), getFlag(fs(1), 0x02), getFlag(fs(1), 0x04),
+                                       getFlag(fs(1), 0x08), getFlag(fs(1), 0x10), getFlag(fs(1), 0x40),
+                                       getFlag(fs(1), 0x80), getFlag(fs(0), 0x10), getFlag(fs(0), 0x40))
+    }
+
+    object FieldFlags {
+      def apply(fs: Array[Byte]) = new FieldFlags(fs)
+    }
+
+    case class MethodFlags(accPublic: Boolean,    accPrivate: Boolean,   accProtected: Boolean,
+                           accStatic: Boolean,    accFinal: Boolean,     accSynchronized: Boolean,
+                           accBridge: Boolean,    accVarargs: Boolean,   accNative: Boolean,
+                           accAbstract: Boolean,  accStrict: Boolean,    accSynthetic: Boolean) {
+
+      def this(fs: Array[Byte]) = this(getFlag(fs(1), 0x01), getFlag(fs(1), 0x02), getFlag(fs(1), 0x04),
+                                       getFlag(fs(1), 0x08), getFlag(fs(1), 0x10), getFlag(fs(1), 0x20),
+                                       getFlag(fs(1), 0x40), getFlag(fs(1), 0x80), getFlag(fs(0), 0x01),
+                                       getFlag(fs(0), 0x04), getFlag(fs(0), 0x08), getFlag(fs(0), 0x10))
+    }
+
+    object MethodFlags {
+      def apply(fs: Array[Byte]) = new MethodFlags(fs)
+    }
+
+    case class ClassFlags(accPublic: Boolean,     accFinal: Boolean,    accSuper: Boolean,
+                          accInterface: Boolean,  accAbstract: Boolean, accSynthetic: Boolean,
+                          accAnnotation: Boolean, accEnum: Boolean){
+
+      def this(fs: Array[Byte]) = this(getFlag(fs(1), 0x01), getFlag(fs(1), 0x10), getFlag(fs(1), 0x20),
```

Comment:
Let's format this function call, and others like it, as 

```scala
def this(fs: Array[Byte]) = this(
  getFlag(fs(1), 0x01), getFlag(fs(1), 0x10), getFlag(fs(1), 0x20),
  getFlag(fs(0), 0x02), getFlag(fs(0), 0x04), getFlag(fs(0), 0x10),
  getFlag(fs(0), 0x20), getFlag(fs(0), 0x40)
)
```

The current convention in the codebase is 

- parameter lists at definition-sites and vertical-aligned with closing-paren on same line, 
- parameter lists at call-sites are indented two-spaces, with closing paren on new line un-indented

It's a bit of an idiosyncratic formatting style which is basically what IntelliJ defaults to, but let's maintain it until we can get something like scalafmt to take over formatting responsibilities.